NineChronicles.Standalone/ActionEvaluationPublisher.cs

```diff
@@ -167,5 +182,66 @@ public override async Task StopAsync(CancellationToken cancellationToken)
             await _client?.DisposeAsync();
             await base.StopAsync(cancellationToken);
         }
+
+        private bool ContainsAddressToBroadcast(ActionBase.ActionEvaluation<ActionBase> ev)
+        {
+            if (_addressesToSubscribe is null ||
+                !_addressesToSubscribe.Any())
+            {
+                return false;
+            }
+
+            if (ev.Signer.Equals(_agentAddress) ||
+                ev.OutputStates.UpdatedFungibleAssets.ContainsKey(_agentAddress))
+            {
+                return true;
+            }
+
+            var updatedAddresses = ev.OutputStates.UpdatedAddresses;
+            foreach (var address in _addressesToSubscribe)
+            {
+                if (updatedAddresses.Contains(address))
+                {
+                    return true;
+                }
+            }
+            
+            return false;
+        }
+
+        private void ResetAddressesToSubscribe(ActionBase.ActionEvaluation<ActionBase> ev)
+        {
+            if (!(ev.Action is CreateAvatar _) &&
+                !(ev.Action is CreateAvatar2 _))
+            {
+                return;
+            }
+
+            if (!ev.Signer.Equals(_agentAddress))
+            {
+                return;
+            }
+            
+            var chainAgentState = ev.OutputStates.GetState(_agentAddress);
+            if (chainAgentState is null)
+            {
+                return;
+            }
+            
+            var agentState = new AgentState((Bencodex.Types.Dictionary) chainAgentState);
+            ResetAddressesToSubscribe((_agentAddress, agentState.avatarAddresses.Values.ToList()));
+        }
+        
+        private void ResetAddressesToSubscribe((Address agentAddress, List<Address> avatarAddresses) tuple)
+        {
+            Log.Debug($"ResetAddressesToSubscribe() invoked. {tuple.agentAddress} {tuple.avatarAddresses?.Count ?? 0}");
+            _agentAddress = tuple.agentAddress;
+            _addressesToSubscribe = new List<Address> {_agentAddress};
```

Comment:
i think we can just subscribe `_agentAddress`.

Reply:
I will leave only `_agentAddress` through next PR. @ipdae